internal/servicelog/formatter.go

```diff
@@ -52,7 +54,83 @@ func NewFormatWriter(dest io.Writer, serviceName string) io.Writer {
 	}
 }
 
-func (f *formatter) Write(p []byte) (nn int, ee error) {
+type TrimWriter struct {
+	dest     io.Writer
+	re       *regexp.Regexp
+	buf      []byte
+	postTrim bool
+	mut      sync.Mutex
+}
+
+func NewTrimWriter(dest io.Writer, re *regexp.Regexp) *TrimWriter {
+	return &TrimWriter{
+		dest: dest,
+		re:   re,
+	}
+}
+
+func (w *TrimWriter) Flush() error {
+	w.mut.Lock()
+	defer w.mut.Unlock()
+
+	if len(w.buf) == 0 {
+		return nil
+	}
+
+	loc := w.re.FindIndex(w.buf)
```

Comment:
It seems like this is the same trimming code as below -- can we combine into an unexported helper method?

Reply:
Good catch - refactored.